code/datums/looping_sounds/_looping_sound.dm

```diff
@@ -33,61 +33,62 @@
 	var/extra_range = 0
 	var/falloff_exponent
 	var/timerid
+	var/looping = FALSE
 	var/falloff_distance
+	var/skip_starting_sounds = FALSE
 
-/datum/looping_sound/New(list/_output_atoms=list(), start_immediately=FALSE, _direct=FALSE)
+/datum/looping_sound/New(_parent, start_immediately=FALSE, _direct=FALSE, _skip_starting_sounds = FALSE)
 	if(!mid_sounds)
 		WARNING("A looping sound datum was created without sounds to play.")
 		return
 
-	output_atoms = _output_atoms
+	set_parent(_parent)
 	direct = _direct
+	skip_starting_sounds = _skip_starting_sounds
 
 	if(start_immediately)
 		start()
 
 /datum/looping_sound/Destroy()
-	stop()
-	output_atoms = null
+	stop(TRUE)
 	return ..()
 
-/datum/looping_sound/proc/start(atom/add_thing)
-	if(add_thing)
-		output_atoms |= add_thing
+/datum/looping_sound/proc/start(on_behalf_of)
+	if(on_behalf_of)
+		set_parent(on_behalf_of)
 	if(timerid)
 		return
 	on_start()
 
-/datum/looping_sound/proc/stop(atom/remove_thing)
-	if(remove_thing)
-		output_atoms -= remove_thing
+/datum/looping_sound/proc/stop(null_parent)
+	if(null_parent)
+		set_parent(null)
 	if(!timerid)
 		return
 	on_stop()
 	deltimer(timerid, SSsound_loops)
 	timerid = null
+	looping = FALSE
 
 /datum/looping_sound/proc/sound_loop(starttime)
 	if(max_loops && world.time >= starttime + mid_length * max_loops)
 		stop()
 		return
 	if(!chance || prob(chance))
 		play(get_sound(starttime))
-	if(!timerid)
-		timerid = addtimer(CALLBACK(src, .proc/sound_loop, world.time), mid_length, TIMER_CLIENT_TIME | TIMER_STOPPABLE | TIMER_LOOP, SSsound_loops)
+	if(!looping)
```

Comment:
The timer handles looping and the timerid being filled determined whether it was ongoing, why this new var?

Reply:
Because I'm using the timer id as the id of the current timer in the step, instead of as the id of the looping timer. That's one of the behavior changes I needed to do to support changing between sound loops without stacking them

Reply:
Discussed on discord, I pulled the looping stuff out to a separate proc, but I can't totally get rid of the looping var without changing behavior